logstash-core/lib/logstash/util/wrapped_synchronous_queue.rb

```diff
@@ -190,28 +192,29 @@ def initialize(queue, size, wait)
         @size = size
         @wait = wait
 
-        @originals = Hash.new
-
         # TODO: disabled for https://github.com/elastic/logstash/issues/6055 - will have to properly refactor
         # @cancelled = Hash.new
 
+        @originals = java.util.HashSet.new size, 1
         @generated = Hash.new
         @iterating_temp = Hash.new
         @iterating = false # Atomic Boolean maybe? Although batches are not shared across threads
         @acked_batch = nil
       end
 
       def read_next
-        @size.times do |t|
-          event = @queue.poll(@wait)
-          return if event.nil? # queue poll timed out
-
-          @originals[event] = true
+        @queue.queue.drainTo(@originals, @size)
+        if @originals.size == 0 
+          e = @queue.poll(@wait)
```

Comment:
This looks like we no longer poll in a loop anymore, but rather poll once? This would be a departure from our current guarantees AFAICT, what's the rationale here?